src/Documentation/Event-Sourcing/LogConsistencyProviders.md

```diff
@@ -0,0 +1,54 @@
+---
+layout: page
+title: Included Log-Consistency Providers
+---
+
+# Built-In Log-Consistency Providers
+
+The `Microsoft.Orleans.EventSourcing` package includes several log-consistency providers that cover basic scenarios suitable to get started, and allow some extensibility.
+
+
+### Orleans.EventSourcing.**StateStorage**.LogConsistencyProvider
+
+This provider stores *grain state snapshots*, using a standard storage provider that can be configured independently. 
+
+The data that is kept in storage is an object that contains both the grain state (as specified by the first type parameter to `JournaledGrain`) and some meta-data (the version number, amd a special tag that is used to avoid duplication of events when storage accesses fail).
+
+Since the entire grain state is read/written every time we access storage, this provider is not suitable for objects whose grain state is very large. 
+
+This provider does not support `RetrieveConfirmedEvents`: it cannot retrieve the events from storage because the events are not persisted.
```

Comment:
One thing that we can do pretty easily is to combine the approach for the ES storage interface (#2723) with  a configuration option for automatic checkpoints via IStorageProvider. 

Basically, the user (in addition to specifying an event store) also (optionally) specifies a standard storage provider, and gives a configuration option on how often they want check points, e.g. "CheckPointDistance=100".

We can then automatically save these checkpoints - these checkpoints then serve as optimizations to reduce load time, i.e. if not found or not deserializable, we just go back to replaying the full event sequence.